test/kubemark/run-e2e-tests.sh

```diff
@@ -32,21 +32,23 @@ export KUBE_MASTER_URL="https://${KUBE_MASTER_IP}"
 export KUBECONFIG="${ABSOLUTE_ROOT}/test/kubemark/resources/kubeconfig.kubemark"
 export E2E_MIN_STARTUP_PODS=0
 
-if [[ -z "$@" ]]; then
-	ARGS='--ginkgo.focus=[Feature:Performance]'
+if [[ -z "$*" ]]; then
+	ARGS=('--ginkgo.focus=[Feature:Performance]')
 else
-	ARGS=$@
+	ARGS=("$@")
 fi
 
 if [[ "${ENABLE_KUBEMARK_CLUSTER_AUTOSCALER}" == "true" ]]; then
-  ARGS="${ARGS} --kubemark-external-kubeconfig=${DEFAULT_KUBECONFIG}"
+  ARGS+=("--kubemark-external-kubeconfig=${DEFAULT_KUBECONFIG}")
 fi
 
 if [[ -f /.dockerenv ]]; then
 	# Running inside a dockerized runner.
-	go run ./hack/e2e.go -- --check-version-skew=false --test --test_args="--e2e-verify-service-account=false --dump-logs-on-failure=false ${ARGS}"
+	go run ./hack/e2e.go -- --check-version-skew=false --test --test_args="--e2e-verify-service-account=false --dump-logs-on-failure=false ${ARGS[*]}"
 else
 	# Running locally.
- 	ARGS=$(echo $ARGS | sed 's/\[/\\\[/g' | sed 's/\]/\\\]/g')
-	${KUBE_ROOT}/hack/ginkgo-e2e.sh "--e2e-verify-service-account=false" "--dump-logs-on-failure=false" $ARGS
+	for ((i=0; i < ${ARGS[@]}; i++)); do
+	  ARGS[$i]="$(echo "ARGS[$i]" | sed -e 's/\[/\\\[/g' -e 's/\]/\\\]/g' )"
+	done
+	"${KUBE_ROOT}/hack/ginkgo-e2e.sh" "--e2e-verify-service-account=false" "--dump-logs-on-failure=false" "${ARGS[@]}"
```

Comment:
You use `ARGS[*]` in the `go run` above, but `ARGS[@]` here, is that on purpose?

Reply:
Yes, my thought was the original was it being interpreted as a string since it was a part of:
```
--test_args="--e2e-verify-service-account=false --dump-logs-on-failure=false ${ARGS}"
```

In the 2nd, it was additional arguments being passed and should the be independent quoted arguments.